cvat/apps/dataset_manager/tests/test_rest_api_formats.py

```diff
@@ -182,12 +288,794 @@ def _generate_url_dump_tasks_annotations(self, task_id):
+                        "Datumaro 1.0" # Datumaro 1.0 is not in the list of import format
+                    ]:
+                        self.skipTest("Format is fail")
+                    images = self._generate_task_images(3)
+                    if dump_format_name == "Market-1501 1.0":
+                        task = self._create_task(tasks["market1501"], images)
+                    elif dump_format_name in ["ICDAR Localization 1.0", "ICDAR Recognition 1.0"]:
+                        task = self._create_task(tasks["icdar_localization_and_recognition"], images)
+                    elif dump_format_name == "ICDAR Segmentation 1.0":
+                        task = self._create_task(tasks["icdar_segmentation"], images)
+                    else:
+                        task = self._create_task(tasks["main"], images)
+                    task_id = task["id"]
+                    if dump_format_name in [
+                        "MOT 1.1", "MOTS PNG 1.0", \
+                        "PASCAL VOC 1.1", "Segmentation mask 1.1", \
+                        "TFRecord 1.0", "YOLO 1.1", "ImageNet 1.0", \
+                        "WiderFace 1.0", "VGGFace2 1.0", \
+                    ]:
+                        self._create_annotations(task, dump_format_name, "default")
+                    else:
+                        self._create_annotations(task, dump_format_name, "random")
+
+                    task_ann = TaskAnnotation(task_id)
+                    task_ann.init_from_db()
+                    task_ann_prev_data = task_ann.data
+                    url = self._generate_url_dump_tasks_annotations(task_id)
+
+                    file_zip_name = osp.join(test_dir, f'{test_name}_{dump_format_name}.zip')
+                    data = {
+                        "format": dump_format_name,
+                        "action": "download",
+                    }
+                    self._download_file(url, data, self.admin, file_zip_name)
+                    self.assertEqual(osp.exists(file_zip_name), True)
+
+                    self._remove_annotations(url, self.admin)
+
+                    self._create_annotations(task, "CVAT for images 1.1 many jobs", "default")
+
+                    if dump_format_name == "CVAT for images 1.1" or dump_format_name == "CVAT for video 1.1":
+                        dump_format_name = "CVAT 1.1"
+                    url = self._generate_url_upload_tasks_annotations(task_id, dump_format_name)
+
+                    with open(file_zip_name, 'rb') as binary_file:
+                        self._upload_file(url, binary_file, self.admin)
+                    task_ann = TaskAnnotation(task_id)
+                    task_ann.init_from_db()
+                    task_ann_data = task_ann.data
+                    self.assertEqual(len(task_ann_data["shapes"]), len(task_ann_prev_data["shapes"]))
+
+    def test_api_v1_unit_test_on_normalize_shape_function(self):
```

Comment:
What does this test tests?

Reply:
Moved.

Reply:
Ok, what about other problems with this test? Currently, it only checks for not throwing an exception. It has 2 repeated test cases.

Reply:
Repeated tests fixed and check length of returned values.